Makefile

```diff
@@ -1,11 +1,30 @@
-.PHONY: test unit_tests integration_tests
+DOT := $(shell command -v dot 2> /dev/null)
+
+
 
 build: install
 
 install:
 	pipenv install --dev
 
-test:
+test: unit_tests component_tests
 	pipenv check
 	pipenv run flake8 .
-	pipenv run pytest test/
+
+unit_tests:
+	pipenv run pytest test/unit/
+
+component_tests: docker_build
+	docker-compose up -d ;
+	./test/component/setup_pubsub.sh
+	pipenv run pytest test/component/
```

Comment:
This never brings down the containers after the test run.

Reply:
added 'docker-compose down' at the end of component-tests